src/main/java/io/cdap/plugin/gcp/bigquery/sink/AvroRecordWriter.java

```diff
@@ -58,6 +59,7 @@ public AvroRecordWriter(Schema writerSchema, GenericData dataModel, CodecFactory
     this.compressionCodec = compressionCodec;
     this.outputStream = outputStream;
     this.syncInterval = syncInterval;
+    createFileWriter(Schema.create(Schema.Type.NULL), ByteStreams.nullOutputStream());
```

Comment:
I tried that 
```
if (mAvroFileWriter != null) {
      mAvroFileWriter.close();
    } else if (outputStream != null) {
      outputStream.close();
    }
```
Without any other change 
but that gives the error : 
`Error occurred while importing data to BigQuery 'Error while reading data, error message: The Apache Avro library failed to parse the header with the following error: EOF reached File : abc/xyz/part-r-00001.avro` 

I am unsure of the reason

Reply:
Also initiating like  `createFileWriter(Schema.create(Schema.Type.NULL), outputStream);` ( instead of NOP  ) fails with NPE while creating `mAvroFileWriter` when actual data is written. ( line  90 --> 103 )

Reply:
Basically now we never close output stream, so the file is hanging around, but due to a side effect is probably not accessible by the importer. It still takes memory, thus the OOM. 

With the fix you propose, I don't think anything is changed as the original OutputStream is not closed anyway, so OOM will be there.

The simple `outputStream.close()` won't work as it would produce an empty file that can't be read by the AVRO reader.

We have 2 options:
 * Don't create the OutputStream if it's not needed. This would require passing a `Supplier<OutputStream>` instead of `OutputStream` to the writer. The result won't change (file won't be created if partition is empty), but the OutputStream won't be created either. In the edge case of all empty partitions, there is going to be no data files at all. I suppose this is how it is now, but it would be great to test if it works / will work after the change
 * Create both OutputStream and DataFileWriter and close in the end. This would create files with proper header, but 0 rows. Here you would also need a proper schema (I believe it's already passed to the  constructor). The downside is creating unneeded files. I would prefer to go first option unless we have issues with empty datasets.

Reply:
Implemented the 1st Supplier approach and it worked :) 

Was able to repro original OOM issue and this code change passed without any issue. 
This also resulted in Faster execution as well ! 

Thanks